src/ZPublisher/HTTPResponse.py

```diff
@@ -433,6 +433,10 @@ def insertBase(self):
                     self.setHeader('content-length', len(self.body))
 
     def isHTML(self, text):
+        try:
```

Comment:
Should be checking for `isintance(text, bytes)` before decoding.

Reply:
You mean it should only decode when text is not bytes?

```python
if not isintance(text, bytes):
    text = text.decode(self.charset)
```

Reply:
Not really fluent in py3 yet. Like this?

```python
    def isHTML(self, text):
        if isintance(text, bytes):
            try:
                text = text.decode(self.charset)
            except UnicodeDecodeError:
                pass
        text = text.lstrip()
        # Note that the string can be big, so text.lower().startswith()
        # is more expensive than s[:n].lower().
        if (text[:6].lower() == '<html>' or
                text[:14].lower() == '<!doctype html'):
            return True
        if text.find('</') > 0:
            return True
        return False
```